plugin/route53/route53.go

```diff
@@ -99,6 +120,74 @@ func (h *Route53) Run(ctx context.Context) error {
 	return nil
 }
 
+func (h *Route53) updateAliases(ctx context.Context) error {
+	errc := make(chan error)
+	defer close(errc)
+
+	for zName, z := range h.zones {
+		go func(zName string, z []*zone) {
+			var err error
+			defer func() {
+				errc <- err
+			}()
+			var rrs []dns.RR
+			var nameserver string
+			for _, hostedZone := range z {
+				for name, alias := range hostedZone.aliases {
```

Comment:
There's a whole lot of nesting going on here. It makes sense to split this up into sub-routines to make this more comprehensible.

Reply:
I've refactored this part a bit so now it should be simpler and easier to read. Let me know what you think.